keentools_facebuilder/interface/updater.py

```diff
@@ -48,6 +53,15 @@ class FBUpdater:
     # _response = mock_response()  # Mock for testing (1/2)
     _parsed_response_content = None
 
+    @classmethod
+    def is_active(cls):
+        return cls.has_response() and (not updates_downloaded() or
```

Comment:
is_active() is a frequent called method by UI-polls
But second part of this condition is a heavy file-reading and files-exists checking function
It would be better to separate this heavy code to a panel drawing section at least
I think some status flag is needed here